compiler/modules/types/core.bal

```diff
@@ -1357,6 +1357,7 @@ final CellAtomicType CELL_ATOMIC_INNER_RO = { ty: INNER_READONLY, mut: CELL_MUT_
 final MappingAtomicType MAPPING_ATOMIC_INNER = { names: [], types: [], rest: CELL_SEMTYPE_INNER };
 final ListAtomicType LIST_ATOMIC_INNER = { members: { initial: [], fixedLength: 0 }, rest: CELL_SEMTYPE_INNER };
 final ListAtomicType LIST_ATOMIC_MAPPING = { members: {initial: [], fixedLength: 0 }, rest: CELL_SEMTYPE_INNER_MAPPING };
+final FunctionAtomicType FUNCTION_ATOMIC_TOP = [NEVER, ANY];
```

Comment:
@jclark I don't think our original implementation was wrong for function types since the top type for the parameter part is not a list type. This is because parameters are contravariant, thus top type must be a subtype of all parameter list types, which is `NEVER`. But it might still be worth it to make it explicit to avoid any confusion in the future.

Reply:
I see your point. I suggest you comment FunctionAtomicType accordingly.